cloudSupport/src/main/scala/cromwell/cloudsupport/gcp/auth/GoogleAuthMode.scala

```diff
@@ -81,14 +71,26 @@ sealed trait GoogleAuthMode {
 
   def requiresAuthFile: Boolean = false
 
+  // changing this is for testing only. all the inheritance is otherwise locked down.
+  private[auth] var credentialValidation: (Credentials => Unit) = credentials => credentials.refresh()
+
   protected def validateCredential(credential: Credentials) = {
-    Try(credential.refresh()) match {
+    Try(credentialValidation(credential)) match {
       case Failure(ex) => throw new RuntimeException(s"Google credentials are invalid: ${ex.getMessage}", ex)
       case Success(_) => credential
     }
   }
 }
 
+case object MockAuthMode extends GoogleAuthMode {
```

Comment:
Is this used only in test code? if so would it make more sense to live over in test-land?

Reply:
That would be nice, but I don't think that works with `sealed trait`s. In this PR I just moved the class from an inner-class to a top-level-class.